tests/utils/runner/runner.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return fmt.Errorf("failed to parse ANR log level: %w", err)
+	}
+	logFactory := logging.NewFactory(logging.Config{
+		DisplayLevel: logLevel,
+		LogLevel:     logLevel,
+	})
+	zapLog, err := logFactory.Make("main")
+	if err != nil {
+		return fmt.Errorf("failed to make client log: %w", err)
+	}
+
+	n.anrClient, err = runner_sdk.New(runner_sdk.Config{
+		Endpoint:    "0.0.0.0:12352",
+		DialTimeout: 10 * time.Second,
+	}, zapLog)
+	if err != nil {
+		return fmt.Errorf("failed to start ANR client: %w", err)
+	}
+
+	return nil
+}
+
+// initServer starts the ANR server if it is not populated
+func (n *NetworkManager) initServer() error {
+	if n.anrServer != nil {
+		return nil
+	}
+
+	_, err := n.startServer(context.Background())
+	return err
+}
+
+// initClient starts an ANR client if it not populated
+func (n *NetworkManager) initClient() error {
+	if n.anrClient != nil {
+		return nil
+	}
+
+	return n.startClient()
+}
+
+// init starts the ANR server and client if they are not yet populated
+func (n *NetworkManager) init() error {
+	if err := n.initServer(); err != nil {
+		return err
+	}
+	return n.initClient()
+}
+
+// StartDefaultNetwork constructs a default 5 node network.
+func (n *NetworkManager) StartDefaultNetwork(ctx context.Context) (<-chan struct{}, error) {
+	if err := n.init(); err != nil {
+		return nil, err
+	}
+
+	log.Info("Sending 'start'", "AvalancheGoExecPath", n.ANRConfig.AvalancheGoExecPath)
+
+	// Start cluster
+	resp, err := n.anrClient.Start(
+		ctx,
+		n.ANRConfig.AvalancheGoExecPath,
+		runner_sdk.WithPluginDir(n.ANRConfig.PluginDir),
+		runner_sdk.WithGlobalNodeConfig(n.ANRConfig.GlobalNodeConfig),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start ANR network: %w", err)
+	}
+	log.Info("successfully started cluster", "RootDataDir", resp.ClusterInfo.RootDataDir, "Subnets", resp.GetClusterInfo().GetSubnets())
+	return n.done, nil
+}
+
+// SetupNetwork constructs blockchains with the given [blockchainSpecs] and adds them to the network manager.
+// Uses [execPath] as the AvalancheGo binary execution path for any started nodes.
+// Note: this assumes that the default network has already been constructed.
+func (n *NetworkManager) SetupNetwork(ctx context.Context, execPath string, blockchainSpecs []*rpcpb.BlockchainSpec) error {
+	if err := n.init(); err != nil {
+		return err
+	}
+	sresp, err := n.anrClient.CreateBlockchains(
+		ctx,
+		blockchainSpecs,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create blockchains: %w", err)
+	}
+
+	// TODO: network runner health should imply custom VM healthiness
+	// or provide a separate API for custom VM healthiness
+	// "start" is async, so wait some time for cluster health
+	log.Info("waiting for all VMs to report healthy", "VMID", evm.ID)
+	for {
+		v, err := n.anrClient.Health(ctx)
+		log.Info("Pinged CLI Health", "result", v, "err", err)
+		if err != nil {
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		// TODO: clean this up
```

Comment:
this logic seems weird. if err != nil we will continuously wait, when would we enter the second code block?